src/elements/content-sidebar/versions/VersionsItem.js

```diff
@@ -82,9 +85,18 @@ const VersionsItem = ({
     const versionInteger = versionNumber ? parseInt(versionNumber, 10) : 1;
     const versionTime = restoredAt || trashedAt || createdAt;
     const versionTimestamp = versionTime && new Date(versionTime).getTime();
-    const versionUserName = selectors.getVersionUser(version).name || (
-        <FormattedMessage {...messages.versionUserUnknown} />
-    );
+    const versionUser = selectors.getVersionUser(version);
+    let versionUserName = versionUser.name || <FormattedMessage {...messages.versionUserUnknown} />;
+    if (uploader_display_name) {
+        if (versionUser.id === PLACEHOLDER_USER.id) {
+            versionUserName =
+                uploader_display_name === FILE_REQUEST ? (
```

Comment:
I'm seeing some repetition between components. It seems like we're trying to do two things: 

1. Determine if the user is anonymous and, if so, substitute the blank name for `uploader_display_name` if available
2. Translate the `uploader_display_name` if it's a generic server-generated value (I assume)

It seems like the code for goal 1 should live in `selectors.getVersionUser` with a small helper method to extend/modify PLACEHOLDER_USER as needed.

It seems like the code for goal 2 should live in each component, since they each use different implementations to internationalize content (`injectIntl` vs `FormattedMessage`).